### PR TITLE
Fix go lint error for cache directory

### PR DESCRIPTION
**Pull request by RainbowMango**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
1. Fix lint errors for the whole directory `staging/src/k8s.io/client-go/tools/cache`;
2. Remove `staging/src/k8s.io/client-go/tools/cache` from `.golint_failures`;
3. Fix some typo from comments.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Review by cheftako — COMMENTED** [review]
Thanks for helping with our quality.
I assume for looking at the changes you are trying to get our lint errors down.
If we don't clear all the errors in the directory and then mark the directory as clean then it is very easy for someone else to (re)introduce more lint errors in the directory.
Can you remove staging/src/k8s.io/client-go/tools/cache from https://github.com/kubernetes/kubernetes/blob/master/hack/.golint_failures#L510 ?

**Comment by RainbowMango** [comment]
@cheftako 

> I assume for looking at the changes you are trying to get our lint errors down.

Lint cannot find these. Actually, these typos are found when I going through the code, not by lint. 
I don't know if there is a way to find all typos for a whole directory.

**Comment by RainbowMango** [comment]
After removing `staging/src/k8s.io/client-go/tools/cache` from `.golint_failures` and run `hack/verify-golint.sh` locally, there are lots of lint errors.

I'm new to kubernetes. I guess the pakage list in `.golint_failures` will ignore golint errors.
Could you please share more information about why `staging/src/k8s.io/client-go/tools/cache` exist in `.golint_failures` ? 

Is it the time for us to clean up these errors? If so, i'd like to provide a PR. Maybe another PR.

```

Errors from golint:
staging/src/k8s.io/client-go/tools/cache/controller.go:82:6: exported type Controller should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:163:1: comment on exported method DeltaFIFO.HasSynced should be of the form "HasSynced ..."
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:392:1: comment on exported method DeltaFIFO.IsClosed should be of the form "IsClosed ..."
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:597:2: exported const Added should have comment (or a comment on this block) or be unexported
staging/src/k8s.io/client-go/tools/cache/delta_fifo.go:600:2: comment on exported const Sync should be of the form "Sync ..."
staging/src/k8s.io/client-go/tools/cache/expiration_cache_fakes.go:36:6: exported type FakeExpirationPolicy should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/expiration_cache_fakes.go:41:1: exported method FakeExpirationPolicy.IsExpired should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/expiration_cache_fakes.go:46:1: exported function NewFakeExpirationStore should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/expiration_cache.go:58:2: struct field Ttl should be TTL
staging/src/k8s.io/client-go/tools/cache/fake_custom_store.go:19:1: comment on exported type FakeCustomStore should be of the form "FakeCustomStore ..." (with optional leading article)
staging/src/k8s.io/client-go/tools/cache/fifo.go:37:5: error var FIFOClosedError should have name of the form ErrFoo
staging/src/k8s.io/client-go/tools/cache/fifo.go:37:5: exported var FIFOClosedError should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/fifo.go:37:21: should omit type error from declaration of var FIFOClosedError; it will be inferred from the right-hand side
staging/src/k8s.io/client-go/tools/cache/fifo.go:69:1: comment on exported function Pop should be of the form "Pop ..."
staging/src/k8s.io/client-go/tools/cache/fifo.go:129:1: comment on exported method FIFO.HasSynced should be of the form "HasSynced ..."
staging/src/k8s.io/client-go/tools/cache/fifo.go:245:1: comment on exported method FIFO.IsClosed should be of the form "IsClosed ..."
staging/src/k8s.io/client-go/tools/cache/heap.go:31:6: exported type LessFunc should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/heap.go:161:1: comment on exported method Heap.BulkAdd should be of the form "BulkAdd ..."
staging/src/k8s.io/client-go/tools/cache/heap.go:254:9: if block ends with a return statement, so drop this else and outdent its block
staging/src/k8s.io/client-go/tools/cache/index.go:78:2: exported const NamespaceIndex should have comment (or a comment on this block) or be unexported
staging/src/k8s.io/client-go/tools/cache/listers.go:33:1: exported function ListAll should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/listers.go:53:1: exported function ListAllByNamespace should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/listers.go:127:1: exported function NewGenericLister should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/mutation_cache.go:45:6: exported type ResourceVersionComparator should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/mutation_detector.go:39:6: exported type CacheMutationDetector should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/mutation_detector.go:39:6: type name will be used as cache.CacheMutationDetector by other packages, and that stutters; consider calling this MutationDetector
staging/src/k8s.io/client-go/tools/cache/mutation_detector.go:44:1: exported function NewCacheMutationDetector should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/shared_informer.go:119:6: exported type SharedIndexInformer should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/shared_informer.go:225:1: receiver name c should be consistent with previous receiver name v for dummyController
staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go:305:1: exported function NewThreadSafeStore should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/undelta_store.go:45:1: exported method UndeltaStore.Add should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/undelta_store.go:53:1: exported method UndeltaStore.Update should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/undelta_store.go:61:1: exported method UndeltaStore.Delete should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/undelta_store.go:69:1: exported method UndeltaStore.Replace should have comment or be unexported
staging/src/k8s.io/client-go/tools/cache/undelta_store_test.go:50:22: should drop = 0 from declaration of var callcount; it is the zero value
staging/src/k8s.io/client-go/tools/cache/undelta_store_test.go:76:22: should drop = 0 from declaration of var callcount; it is the zero value
staging/src/k8s.io/client-go/tools/cache/undelta_store_test.go:113:22: should drop = 0 from declaration of var callcount; it is the zero value
```

**Comment by RainbowMango** [comment]
@cheftako 
Fixed all whole directory golint errors.  Please help to review.

**Comment by deads2k** [comment]
weird that deps failed.  i don't see any dep changes.

/ok-to-test
/lgtm

**Comment by deads2k** [comment]
/hold
needs squash

**Comment by RainbowMango** [comment]
@deads2k 
Squashed. PTAL.

**Comment by RainbowMango** [comment]
Need to remove `hold` label and approve.

File: staging/src/k8s.io/client-go/tools/cache/index.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ func IndexFuncToKeyFuncAdapter(indexFunc IndexFunc) KeyFunc {
 }
 
 const (
+	// NamespaceIndex is the most popular name to a IndexFunc.
```
Comment (cheftako) [inline staging/src/k8s.io/client-go/tools/cache/index.go]: "popular name" seems odd.

"NamespaceIndex" is the lookup name for the most comment index function, which is to index by the namespace field"?

File: staging/src/k8s.io/client-go/tools/cache/listers.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ import (
 // AppendFunc is used to add a matching item to whatever list the caller is using
 type AppendFunc func(interface{})
 
+// ListAll used to list items according selector from Store.
```
Comment (cheftako) [inline staging/src/k8s.io/client-go/tools/cache/listers.go]: Correct but seems hard to parse.

"ListAll calls appendFn with each value retrieved from store which matches the selector"?

**Review by cheftako — COMMENTED** [review]
Couple of nits but
/lgtm

**Comment by cheftako** [comment]
You still need an approval for the kubelet change. You should probably assign one of the approvers from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/OWNERS.

File: staging/src/k8s.io/client-go/tools/cache/index.go (hunk 1 above)
Comment (RainbowMango) [reply staging/src/k8s.io/client-go/tools/cache/index.go]: Done. Thanks.

File: staging/src/k8s.io/client-go/tools/cache/listers.go (hunk 2 above)
Comment (RainbowMango) [reply staging/src/k8s.io/client-go/tools/cache/listers.go]: Done. Thanks.

**Comment by RainbowMango** [comment]
Maybe @liggitt can help to approve this PR as no response from `pkg/kubelet/OWNERS`. :)

**Comment by nikhita** [comment]
/assign @yujuhong 
for approval

**Comment by RainbowMango** [comment]
Seems needs lgtm also :)

**Comment by RainbowMango** [comment]
> go: github.com/gogo/protobuf@v0.0.0-20190410021324-65acae22fc9: unexpected status (https://proxy.golang.org/github.com/gogo/protobuf/@v/v0.0.0-20190410021324-65acae22fc9.info): 410 Gone

/retest
